src/openzeppelin/tests/access/ownable/test_dual_ownable.cairo

```diff
@@ -0,0 +1,203 @@
+    let camel_target = utils::deploy(CamelOwnablePanicMock::TEST_CLASS_HASH, ArrayTrait::new());
+    (
+        DualCaseOwnable {
+            contract_address: snake_target
+            }, DualCaseOwnable {
+            contract_address: camel_target
+        }
+    )
+}
+
+///
+/// Case agnostic methods
+///
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_owner() {
+    let (snake_dispatcher, _) = setup_snake();
+    let (camel_dispatcher, _) = setup_camel();
+    assert(snake_dispatcher.owner() == OWNER(), 'Should return OWNER');
+    assert(camel_dispatcher.owner() == OWNER(), 'Should return OWNER');
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('ENTRYPOINT_NOT_FOUND', ))]
+fn test_dual_no_owner() {
+    let dispatcher = setup_non_ownable();
+    dispatcher.owner();
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Some error', 'ENTRYPOINT_FAILED', ))]
+fn test_dual_owner_exists_and_panics() {
+    let (dispatcher, _) = setup_ownable_panic();
+    dispatcher.owner();
+}
+
+///
+/// snake_case target
+///
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_transfer_ownership() {
+    let (dispatcher, target) = setup_snake();
+    set_contract_address(OWNER()); // Bug with test-runner
+    dispatcher.transfer_ownership(NEW_OWNER());
+    assert(target.owner() == NEW_OWNER(), 'Should be new owner');
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('ENTRYPOINT_NOT_FOUND', ))]
+fn test_dual_no_transfer_ownership() {
+    let dispatcher = setup_non_ownable();
+    dispatcher.transfer_ownership(NEW_OWNER());
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Some error', 'ENTRYPOINT_FAILED', ))]
+fn test_dual_transfer_ownership_exists_and_panics() {
+    let (dispatcher, _) = setup_ownable_panic();
+    dispatcher.transfer_ownership(NEW_OWNER());
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_renounce_ownership() {
+    let (dispatcher, target) = setup_snake();
+    set_contract_address(OWNER()); // Bug with test-runner
+    dispatcher.renounce_ownership();
+    assert(target.owner().is_zero(), 'Should be zero');
+}
+
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('ENTRYPOINT_NOT_FOUND', ))]
+fn test_dual_no_renounce_ownership() {
+    let dispatcher = setup_non_ownable();
+    dispatcher.renounce_ownership();
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Some error', 'ENTRYPOINT_FAILED', ))]
+fn test_dual_renounce_ownership_exists_and_panics() {
+    let (dispatcher, _) = setup_ownable_panic();
+    dispatcher.renounce_ownership();
+}
+
+///
+/// camelCase target
+///
+
+#[ignore] // Bug with test-runner
+#[test]
+#[available_gas(2000000)]
+fn test_dual_transferOwnership() {
+    let (dispatcher, target) = setup_camel();
+    set_contract_address(OWNER()); // Bug with test-runner
+    dispatcher.transfer_ownership(NEW_OWNER());
+    assert(target.owner() == NEW_OWNER(), 'Should change ownership');
```

Comment:
should we normalize the assert error string? the snake case analogous one is different